### ISSUE TITLE
Linking to mkdocs site takes me to the homepage

### ISSUE THREAD

**Issue by mclaireh** [mkdocs/mkdocs#2174]
I have been having a problem with trying to link to my site. Sometimes when I copy and paste the URL from a page on my site and then send it to someone it will take them directly to the homepage as opposed to the page the URL came from. Has anyone else had this problem and know how to fix it? I've tried looking for answers online but haven't found anything helpful.

**Comment by facelessuser** [comment]
You would have to provide some more information. I give links to pages of my mkdocs site all the time with no issues. The site uses those links to navigate internally, so if the links were wrong, you wouldn't be able to navigate the site at all.

How do you host your site? On GitHub pages? Something else?

What is an example link? Where do you want it to go?

Right now it is impossible to debug as we only have a vague description with no way for anyone to reproduce. If you can provide more details, maybe someone can help guide you in the right direction.

**Comment by mclaireh** [comment]
I am hosting the site on gitlab pages. The links used to work but just recently started giving us problems. It is for a documentation site that is only available to employees so you won't be able to pull up the site, but I have provided an example link below. This link should take us to a page with a sample payload but is taking us to the homepage. But sometimes it will take us to the right page. It doesn't always default to the homepage, only sometimes. And it only happens when copying the link and sending it to someone. The links on the actual site that have been generated works fine. Let me know if there is anymore info that I can provide that might help.

Example Link:
https://ourcompanysURL.org/claim_events_wiki/#Consumers/PayloadDetails/SampleInputOutput/SampleInputGatewayPayload/

**Comment by facelessuser** [comment]
Is it only links with `#` in the name? That seems an odd character for titles.

**Comment by mclaireh** [comment]
I have looked into my mkdocs.yml file and I can't seem to get rid of the # in the link. Here is an example from the file:
```
- Consumers:
    - Intake Process:
      - Intake Info: Consumers/IntakeProcess/claimsCloudEventingIntakeInformation.md
      - NPI/SPI Info: Consumers/IntakeProcess/npi-spi.md
```
And those links respectively show as:
http://127.0.0.1:8000/#Consumers/IntakeProcess/claimsCloudEventingIntakeInformation/
http://127.0.0.1:8000/#Consumers/IntakeProcess/npi-spi/

I've tried adding a / before Consumers and a ./ before Consumers but the # sign still showed up. There are no #'s in any of the folders under docs or in any of the titles in mkdocs.yml so I'm not sure what's going on with it.
